ivy_lint/formatters/function_ordering.py

```diff
@@ -534,6 +543,106 @@ def sort_key(item: Tuple[str, ast.AST]) -> Tuple[float, int, str]:
 
         reordered_code = black.format_str(reordered_code, mode=black.Mode())
 
+        if extended:
+            return self._extended_rearrange_functions_and_classes(original_source_code, reordered_code, sort_key)
+        return reordered_code
+
+    def _extended_rearrange_functions_and_classes(self, original_source_code: str, reordered_code: str,
+                                                  sort_key) -> str:
+        """
+        Extends the _rearrange_functions_and_classes function to cater to both ivy and stateful files.
+
+        This method utilizes the reordered code from _rearrange_functions_and_classes ensuring that classes, helpers and
+        assignment ordering is maintained, however it reorders the main functions keeping the organizational structure
+        typically found in ivy and stateful files.
+
+        Parameters
+        ----------
+        original_source_code
+            The original source code to be reordered.
+
+        reordered_code
+            The reordered_code from _rearrange_functions_and_classes.
+
+        sort_key: the sort function from _rearrange_functions_and_classes
+
+        Returns
+        -------
+        The reordered source code.
+        """
+        section_headers = [
```

Comment:
@vedpatwardhan 
Are these the only possible section headers in the backend?

Reply:
Yes, I checked through the files.

Reply:
They aren't the only possible section headers (like new ones might get added) but these are the ones in the files currently

Reply:
Oh I see, my bad. I guess I was answering if these are the ones in the files currently.